hw/ip/csrng/doc/_index.md

```diff
@@ -91,3 +91,391 @@ This document aims to make the distinction between physical entropy and factory
+Additionally, the ctr_drbg_gen block also uses the update block and the block_encrypt block.
+To keep resources to a minimum, both of these blocks have arbiters to allow sharing between the ctr_drbg_cmd and ctr_drbg_gen blocks.
+The command field called `ccmd` (for current command) is sent along the pipeline to not only identify the command, but is also reused as a routing tag for the arbiters to use when returning the block response.
+
+Once the command has traversed through all of the ctr_drbg blocks, the result will eventually land into the state_db block.
+This block will hold the instance state for each application interface.
+The specific state information held in the instance is documented below.
+If the command was a `generate` command, the genbits data word will be returned to the requesting `cmd_stage` block.
+Finally, an `ack` response and status will be returned to the application interface once the command has been completely processed.
+
+
+## Block Diagram
+
+![CSRNG Block Diagram](csrng_blk_diag.svg)
+
+## Hardware Interfaces
+
+ {{< hwcfg "hw/ip/csrng/data/csrng.hjson" >}}
+
+
+## Design Details
+
+#### Non-blocking Commands
+Regarding command processing, all commands process immediately except for the generate command.
+The command generate length coutn (`glen`) is kept in the `cmd_stage` block.
+When the state_db block issues an ack to the `cmd_stage` block, the `cmd_stage` block increments an internal counter.
+This process repeats until the `glen` field value has been matched.
+Because each request is pipelined, requests from other `cmd_stage` blocks can be processed before the original generate command is completely done.
+This provides some interleaving of commands since a generate command can be programmed to take a very long time.
+
+#### Working State Values
+The state_db has the follow attributes shown in the following table:
+
+<table>
+<caption>State Instance Description</caption>
+<thead>
+  <tr>
+    <th>Bits</th>
+    <th>Name</th>
+    <th>Description</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td>31:0</td>
+    <td>Reseed Counter</td>
+    <td> Value required by NIST to be held in the state instance.
+    </td>
+  </tr>
+  <tr>
+    <td>159:32</td>
+    <td>V</td>
+    <td> Value required by NIST to be held in the state instance, and is of size BlockLen.
+    </td>
+  </tr>
+  <tr>
+    <td>543:160</td>
+    <td>Key</td>
+    <td> Value required by NIST to be held in the state instance, and is of size SeedLen.
+    </td>
+  </tr>
+  <tr>
+    <td>544</td>
+    <td>Status</td>
+    <td> Set when instantiated.
+    </td>
+  </tr>
+  <tr>
+    <td>545</td>
+    <td>Compliance</td>
+    <td> Set when FIPS/CC compliant entropy was used to seed this instance.
+    </td>
+  </tr>
+</table>
+
+
+#### AES Cipher
+The `block_encrypt` block is where the `aes_cipher_core` block is located. 
+This is the same block used in the AES design.
```

Comment:
i think it would be good to mention this is the unmasked version, just for clarity.

Reply:
Added.